### PR TITLE
Remove some unnecessary link annotations

### PR DESCRIPTION
**Pull request by straight-shoota**
On musl, `librt`, `libm`, and `libpthread` are integrated directly into `libc`. It's unnecessary to link them separately. https://wiki.musl-libc.org/faq.html#Q:-lib(m|pthread|crypt).a/so-are-empty?

Also, we don't need to link `libpthread` in GC none. That's only necessary in the Boehm GC bindings because we need to make sure `pthreads` is passed to the linker before `gc`. GC none doesn't need `libgc` nor `pthreads`.

### CODE REVIEW

File: src/math/libm.cr [hunk 1]
```diff
@@ -1,8 +1,10 @@
+# MUSL: On musl systems, libm is empty. The entire library is already included in libc.
+# The empty library is only available for POSIX compatibility. We don't need to link it.
 # Interpreter: On GNU systems, libm.so is typically a GNU ld script which adds
```
Comment (Sija) [inline src/math/libm.cr:3]:
```suggestion
#
# Interpreter: On GNU systems, libm.so is typically a GNU ld script which adds
```

File: src/gc/boehm.cr [hunk 2]
```diff
@@ -2,7 +2,11 @@
   require "crystal/rw_lock"
 {% end %}
 
-{% unless flag?(:win32) %}
+# MUSL: On musl systems, libpthread is empty. The entire library is already included in libc.
+# The empty library is only available for POSIX compatibility. We don't need to link it.
+# OTHER: On other systems, we add the linker annotation here to make sure libpthread is loaded
```
Comment (Sija) [inline src/gc/boehm.cr:7]:
```suggestion
#
# OTHERS: On other systems, we add the linker annotation here to make sure libpthread is loaded
```

File: src/crystal/system/unix/lib_event2.cr [hunk 3]
```diff
@@ -1,6 +1,8 @@
 require "c/netdb"
 
-{% if flag?(:linux) %}
+# MUSL: On musl systems, libprt is empty. The entire library is already included in libc.
```
Comment (oprypin) [inline src/crystal/system/unix/lib_event2.cr:3]: Is "libprt" a typo?
```suggestion
# MUSL: On musl systems, librt is empty. The entire library is already included in libc.
```

**Review by oprypin — APPROVED** [review]

File: src/crystal/system/unix/lib_event2.cr (hunk 3 above)
Comment (straight-shoota) [reply src/crystal/system/unix/lib_event2.cr:3]: Yes, thanks 😄

**Pull request merged**